### PR TITLE
[4.0] Component and Plugin libraries prepared statements

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

Adds prepared statements (whereIn) and cleans up queries in `Joomla\CMS\Component` and `Joomla\CMS\Plugin` namespaces.

### Testing Instructions

Test that plugins and component still work, e.g. you can still login and can still access components.

### Expected result

Works like before.

### Documentation Changes Required

No.

### CODE REVIEW

File: libraries/src/Plugin/PluginHelper.php [hunk 1]
```diff
@@ -266,34 +266,38 @@ protected static function load()
...
+				->order($db->quoteName('ordering'));
 			$db->setQuery($query);
 
 			return $db->loadObjectList();
 		};
 
 		try
 		{
-			static::$plugins = $cache->get($loader, array(), md5($levels), false);
+			static::$plugins = $cache->get($loader, array(), md5(implode(',', $levels)), false);
```
Comment (Quy) [inline libraries/src/Plugin/PluginHelper.php]:
```suggestion
			static::$plugins = $cache->get($loader, [], md5(implode(',', $levels)), false);
```

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 81d57012d8dab0a0551f812ea55aa9c1a3fb56f8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26012">issues.joomla.org/tracker/joomla-cms/26012</a>.</sub>

**Comment by richard67** [comment]
Code review shows me it is correct and aligns queries to new J4 style but I do not see anything with prepared statements, so either I am blind or the title of this PR is a bit misleading.

**Comment by alikon** [comment]
stricly speaking you are right, it's more like a cleanup

**Comment by richard67** [comment]
Code review looks good, will check soon if it breaks something (what I don't expect).

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 81d57012d8dab0a0551f812ea55aa9c1a3fb56f8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26012">issues.joomla.org/tracker/joomla-cms/26012</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26012">issues.joomla.org/tracker/joomla-cms/26012</a>.</sub>

**Comment by richard67** [comment]
@SharkyKZ I was so free to change title and description because there was nothing to be prepared statement. Change back if you are not ok with it.

**Comment by alikon** [comment]
https://github.com/joomla/joomla-cms/pull/26012/files#diff-4a21854947e3cf421d4bb5d0a805271fR291
can be considered as prepared statement, but who cares , better if this will be merged asasp

**Comment by richard67** [comment]
You are right, the whereIn is prepared. I change back title. Sorry @SharkyKZ .

File: libraries/src/Plugin/PluginHelper.php [hunk 2]
```diff
@@ -266,34 +266,38 @@ protected static function load()
...
-				->where('type = ' . $db->quote('plugin'))
-				->where('state IN (0,1)')
-				->where('access IN (' . $levels . ')')
-				->order('ordering');
+				->from($db->quoteName('#__extensions'))
+				->where(
+					[
+						$db->quoteName('enabled') . ' = 1',
+						$db->quoteName('type') . ' = ' . $db->quote('plugin'),
+						$db->quoteName('state') . ' IN (0,1)',
```
Comment (HLeithner) [inline libraries/src/Plugin/PluginHelper.php:288]:
```suggestion
						$db->quoteName('state') . ' IN (0, 1)',
```

Comment (SharkyKZ) [reply libraries/src/Plugin/PluginHelper.php:288]: We normally don't use spaces with `IN()`. So it's fine as it is, I think.

Comment (HLeithner) [reply libraries/src/Plugin/PluginHelper.php:288]: ok

**Comment by HLeithner** [comment]
thanks for the patch.

**Pull request merged**